src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -11900,32 +11900,23 @@ HRESULT CEEJitInfo::getMethodBlockCounts (
 
 #ifdef FEATURE_PGO
 
-    // For now, only return the info for the method being jitted.
-    // Will need to fix this to gain access to pgo data for inlinees.
     MethodDesc* pMD = (MethodDesc*)ftnHnd;
-
-    if (pMD == m_pMethodBeingCompiled)
+    unsigned codeSize = 0;
+    if (pMD->IsDynamicMethod())
     {
-        unsigned codeSize = 0;
-        if (pMD->IsDynamicMethod())
-        {
-            unsigned stackSize, ehSize;
-            CorInfoOptions options;
-            DynamicResolver * pResolver = m_pMethodBeingCompiled->AsDynamicMethodDesc()->GetResolver();
-            pResolver->GetCodeInfo(&codeSize, &stackSize, &options, &ehSize);
-        }
-        else
-        {
-            codeSize = m_ILHeader->GetCodeSize();
-        }
-
-        hr = PgoManager::getMethodBlockCounts(pMD, codeSize, pCount, pBlockCounts, pNumRuns);
+        unsigned stackSize, ehSize;
+        CorInfoOptions options;
+        DynamicResolver * pResolver = m_pMethodBeingCompiled->AsDynamicMethodDesc()->GetResolver();
+        pResolver->GetCodeInfo(&codeSize, &stackSize, &options, &ehSize);
     }
-    else
+    else if (pMD->HasILHeader())
     {
-        hr = E_NOTIMPL;
+        COR_ILMETHOD_DECODER decoder(pMD->GetILHeader());
+        codeSize = decoder.GetCodeSize();
```

Comment:
It feels like there is some opportunity for a race condition here while interacting with Profiler Rejit, but otherwise this looks good. I'm going to mark this as approved, but we may need a conversation with @davmason to make sure this is safe.

Reply:
Agree rejitting could cause troubles. I wonder this is a broader issue -- we tend to query properties of inlinees over time, and not all in one go.

I actually have the IL size on the jit side so could pass the information down, if needed.

Reply:
I suspect the broader issue is a thing, but given the rarity of rejit we would never find the issue in our testing.

Reply:
Inlining has always ignored profiler modified IL, whether it's from SetILFunctionBody or via rejit. When grabbing the IL for inlining the jit grabs it directly from the MethodDesc and doesn't query the ReJITManager or call Module::GetDynamicIL. 

It's always been an unexpected gotcha for profilers that if you rewrite method B and method A inlines B, even if the inlining occurs after the IL rewrite it will inline the original IL. I never knew the reason for this, perhaps it's intentional because of the way the jit requests inlinee data. Starting in 3.0 inlining is explicitly blocked for any rejit methods. I added logic to CEEInfo::canInline that will return INLINE_FAIL if the method's active IL has been modified with rejit.

Reply:
Ok, thanks.

It might be worth tracing out the fuller picture and rationale somewhere, if it's not written down.